Fabric 1.17

quiqueck opened the pull request:
Some fixes for 1.17.

There are two main Problems left:
- [x]  `IglooGeneratorMixin.addDogsled` -> JUMP Injection fails for some reason
- [x] `TreeFeatureMixin.isDriftwoodTreeGen` -> the references Method `isDirtOrGrass` was removed

quiqueck committed:
Model Fixes

quiqueck committed:
Fixed some mappings

quiqueck committed:
Fixed remaining compiler errors

quiqueck committed:
Fixed Mixins

quiqueck committed:
More mixin fixes

quiqueck committed:
latest versions

quiqueck committed:
Completed Models

quiqueck committed:
Fixed `IglooGeneratorMixin`

quiqueck committed:
Dixed missing `isDirtOrGrass` in `TreeFeature`

quiqueck committed:
Bump Version

quiqueck committed:
1.17.1 update

quiqueck committed:
Fixed server-side crash

Platymemo commented on src/main/java/com/github/platymemo/alaskanativecraft/mixin/client/ClientPlayNetworkHandlerMixin.java:
```diff
@@ -25,17 +25,19 @@
             locals = LocalCapture.CAPTURE_FAILHARD
     )
-    private void onEntitySpawn(EntitySpawnS2CPacket packet, CallbackInfo ci, double x, double y, double z, EntityType<?> type) {
+    private void onEntitySpawn(EntitySpawnS2CPacket packet, CallbackInfo ci, EntityType<?> type) {
```
Actually, due to 1.17 changes, this whole mixin can be removed (thank god)

quiqueck replied:
Verified and removed in 207644bd7bd01e5a41841c25c95562144b62c9a1

Platymemo commented on src/main/java/com/github/platymemo/alaskanativecraft/mixin/client/InGameHudMixin.java:
```diff
@@ -36,25 +37,26 @@
     private void renderSnowGogglesOverlay(MatrixStack matrices, float tickDelta, CallbackInfo ci) {
-        if (this.client.options.getPerspective().isFirstPerson() && this.client.player.inventory.getArmorStack(3).getItem() == AlaskaItems.SNOW_GOGGLES) {
+        if (this.client.options.getPerspective().isFirstPerson() && this.client.player.getInventory().getArmorStack(3).getItem() == AlaskaItems.SNOW_GOGGLES) {
+            //TODO: test
```
I see this TODO, has it been tested?

quiqueck replied:
Actually, no. I seem to have forgotten it, sorry.

However, 7df16722c2fd62282a0e91be0536d049e6fc80de implements a working fix

Platymemo commented on src/main/java/com/github/platymemo/alaskanativecraft/client/renderer/item/HarpoonItemRenderer.java:
```diff
@@ -13,10 +13,13 @@
 
-public enum HarpoonItemRenderer {
-    INSTANCE;
+public class HarpoonItemRenderer {
```
Why was this switched to a class?

quiqueck replied:
I am not sure anymore, probably was necessary at some point and never got changed back. Reverted in 6b269557a7c3b92c487bc0445a85b4bf9bfa86a0

Platymemo commented on src/main/java/com/github/platymemo/alaskanativecraft/client/renderer/entity/feature/KuspukSkirtFeatureRenderer.java:
```diff
@@ -22,11 +24,12 @@
 @Environment(EnvType.CLIENT)
 public class KuspukSkirtFeatureRenderer<T extends LivingEntity, M extends EntityModel<T>> extends FeatureRenderer<T, M> {
-    private final KuspukSkirtModel<T> model = new KuspukSkirtModel<>();
+    private final KuspukSkirtModel<T> model ;
```
```suggestion
    private final KuspukSkirtModel<T> model;
```

Platymemo commented on src/main/java/com/github/platymemo/alaskanativecraft/AlaskaNativeCraftClient.java:
```diff
@@ -34,6 +40,7 @@ public void onInitializeClient() {
         registerItemColors();
         registerBlockRenderLayers();
         registerHarpoonPacket();
+        AlaskaNativeCraftModels.registerEntityModels();
```
I think `registerEntityModels()` should be moved here, to maintain parity with all the other methods. Otherwise, I'd want to break those ones into their respective classes (BE renderers in `AlaskaBlocks`, harpoon packet in `HarpoonEntity`, etc.)

quiqueck replied:
Moved `registerEntityModels()` in 96866f8db0c9766c89d1fa1fa38955d3992bfc4e

Platymemo commented on src/main/java/com/github/platymemo/alaskanativecraft/AlaskaNativeCraftModels.java:
```diff
@@ -0,0 +1,34 @@
+import net.minecraft.util.Identifier;
+
+public class AlaskaNativeCraftModels {
+    public static final EntityModelLayer PTARMIGAN_MODEL = registerMain("ptarmigan");
+    public static final EntityModelLayer DOGSLED_MODEL = registerMain("dogsled");
+    public static final EntityModelLayer HARPOON_MODEL = registerMain("harpoon");
+    public static final EntityModelLayer MOOSE_MODEL = registerMain("moose");
+    public static final EntityModelLayer SEAL_MODEL = registerMain("seal");
+    public static final EntityModelLayer KUSPUK_SKIRT_MODEL = registerMain("kuspuk_skirt");
```
```suggestion
    public static final EntityModelLayer PTARMIGAN = registerMain("ptarmigan");
    public static final EntityModelLayer DOGSLED = registerMain("dogsled");
    public static final EntityModelLayer HARPOON = registerMain("harpoon");
    public static final EntityModelLayer MOOSE = registerMain("moose");
    public static final EntityModelLayer SEAL = registerMain("seal");
    public static final EntityModelLayer KUSPUK_SKIRT = registerMain("kuspuk_skirt");
```
Should follow the same naming scheme that's used in `EntityModelLayers`. That is, just the entity name, no `_MODEL` suffix.

quiqueck replied:
Additional fixes in 32325d32e5b1957cfb67d94b1b2b630e70400700

Platymemo commented on src/main/java/com/github/platymemo/alaskanativecraft/AlaskaNativeCraftModels.java:
```diff
@@ -0,0 +1,34 @@
+package com.github.platymemo.alaskanativecraft;
```
This should probably be in `com.github.platymemo.alaskanativecraft.client.model.entity`

quiqueck replied:
Moved in ecffd3e2a7851f9764856886c105aca42aaa0a78

Platymemo commented on src/main/java/com/github/platymemo/alaskanativecraft/AlaskaNativeCraftModels.java:
```diff
@@ -0,0 +1,34 @@
+import net.fabricmc.fabric.api.client.rendereregistry.v1.EntityModelLayerRegistry;
+import net.minecraft.client.render.entity.model.EntityModelLayer;
+import net.minecraft.util.Identifier;
+
```
```suggestion
import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;

@Environment(EnvType.CLIENT)
```

Platymemo reviewed (changes requested):
Thank you so much for this!!!
I've been very busy for the past month, so seeing this after coming back was really great.
All the comments in the body are questions, suggestions, and requests, so I want to take this comment to be positive lol.
I really appreciate the effort you put into this.

I do want to say, all the commented code in the models can be removed if the models are good. (I couldn't find a good comment point to put this in there)

Platymemo commented:
Also, if you're not willing or able to implement these changes, just lmk and I'll merge this and take care of them myself!

quiqueck commented:
Sorry, just saw your Comments :)
I'll have a look at them and update.

quiqueck committed:
Update src/main/java/com/github/platymemo/alaskanativecraft/client/renderer/entity/feature/KuspukSkirtFeatureRenderer.java

Co-authored-by: Whangd00dle <67663207+Platymemo@users.noreply.github.com>

quiqueck committed:
Update src/main/java/com/github/platymemo/alaskanativecraft/AlaskaNativeCraftModels.java

Co-authored-by: Whangd00dle <67663207+Platymemo@users.noreply.github.com>

quiqueck committed:
Moved to latest Fabric-API

quiqueck committed:
`HarpoonItemRenderer` back as enum

quiqueck committed:
Removed ClientPlayNetworkHandlerMixin

quiqueck committed:
Update src/main/java/com/github/platymemo/alaskanativecraft/AlaskaNativeCraftModels.java

Co-authored-by: Whangd00dle <67663207+Platymemo@users.noreply.github.com>

quiqueck committed:
Removed the *_MODEL* suffix

quiqueck committed:
moved `registerEntityModels`

quiqueck committed:
Changed package for `AlaskaNativeCraftModels`

quiqueck committed:
Fixed SnowGoggles Overlay rendering

quiqueck committed:
Cleaned up some reference comments

quiqueck commented:
Thanks for the feedback so far. I think I addressed all comments. If not, just let me know. 😀 

Btw. please have a closer look at the Models. They are quite complex, so I might have messed things up without noticing.

Platymemo commented on src/main/java/com/github/platymemo/alaskanativecraft/mixin/client/InGameHudMixin.java:
```diff
@@ -23,40 +18,19 @@
-            RenderSystem.enableDepthTest();
-            //RenderSystem.enableAlphaTest();
-            RenderSystem.setShaderColor(1.0F, 1.0F, 1.0F, 1.0F);
+            this.renderOverlay(SNOWGOOGLES_BLUR, 1.0F);
```
Oh that is beautiful, I'm so glad 1.17 has this now

quiqueck replied:
Yeah, that is a great simplification, and makes total sense for the update as they've added two more overlays (spyglas and freezing).

Platymemo reviewed (approved):
LGTM!
